src/NuGet.Status/App_Start/Startup.cs

```diff
@@ -57,13 +57,20 @@ public void Configuration(IAppBuilder app)
                         Authority = _authority,
                         RedirectUri = _redirectUri,
                         PostLogoutRedirectUri = _rootUri,
-                        TokenValidationParameters = new System.IdentityModel.Tokens.TokenValidationParameters
+                        CookieManager = new SystemWebChunkingCookieManager(),
+                        TokenValidationParameters = new Microsoft.IdentityModel.Tokens.TokenValidationParameters
                         {
                             ValidateIssuer = false,
                             RoleClaimType = "roles"
                         },
                         Notifications = new OpenIdConnectAuthenticationNotifications
                         {
+                            RedirectToIdentityProvider = notification =>
+                            {
+                                notification.Options.RedirectUri = new Uri(new Uri(_rootUri), "Authorized").ToString();
```

Comment:
Remove this, you've overriding global state per-request, and the wrong field at that. If you recombine the controllers then you shouldn't need to change any redirects, it will return back to the original endpoint after authenticating.